### PR TITLE
NF-178 - generate EIS request in FE

### PR DESCRIPTION
**Pull request by SteveSugden**

### CODE REVIEW

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/CreateClaimService.scala [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import uk.gov.hmrc.nationalimportdutyadjustmentcentrefrontend.connectors.NIDACConnector
+import uk.gov.hmrc.nationalimportdutyadjustmentcentrefrontend.models.eis.EISCreateCaseRequest
+import uk.gov.hmrc.nationalimportdutyadjustmentcentrefrontend.models.{Claim, CreateClaimResponse}
+import uk.gov.hmrc.nationalimportdutyadjustmentcentrefrontend.services.requests.CreateEISClaimRequest
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class CreateClaimService @Inject() (connector: NIDACConnector)(implicit ec: ExecutionContext) {
+
+  def submitClaim(request: Claim)(implicit hc: HeaderCarrier): Future[CreateClaimResponse] = {
```
Comment (crosslandwa) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/CreateClaimService.scala:32]: I think renaming the passed in `request` to `claim` would make this implementation slightly clearer

```suggestion
  def submitClaim(claim: Claim)(implicit hc: HeaderCarrier): Future[CreateClaimResponse] = {
```

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/CreateClaimService.scala [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    val correlationId = hc.requestId.map(_.value).getOrElse(UUID.randomUUID().toString)
+    val eisRequest: EISCreateCaseRequest = EISCreateCaseRequest(
+      AcknowledgementReference = correlationId.replace("-", ""),
+      ApplicationType = "NIDAC",
+      OriginatingSystem = "Digital",
+      Content = EISCreateCaseRequest.Content(request)
```
Comment (crosslandwa) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/CreateClaimService.scala:39]:
```suggestion
      Content = EISCreateCaseRequest.Content(claim)
```

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/CreateClaimService.scala [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    )
+
+    connector.submitClaim(CreateEISClaimRequest(eisRequest, request.uploads), correlationId)
```
Comment (crosslandwa) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/CreateClaimService.scala:42]:
```suggestion
    connector.submitClaim(CreateEISClaimRequest(eisRequest, claim.uploads), correlationId)
```

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/CreateClaimService.scala (hunk 1 above)
Comment (SteveSugden) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/CreateClaimService.scala:32]: IntelliJ should have done that!  Bah.

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/models/eis/EISCreateCaseRequest.scala [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    FirstName: String,
+    LastName: String
+  )
+
+  object Content {
+    implicit val formats: Format[Content] = Json.format[Content]
+
+    val entryDataFormatter: DateTimeFormatter = DateTimeFormatter.ofPattern("yyyyMMdd")
+
+    def apply(request: Claim): Content =
```
Comment (crosslandwa) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/models/eis/EISCreateCaseRequest.scala:55]: Renaming `request` to `claim` would aide clarity here (in my head i'm conflating "request" with the outgoing HTTP request, but the `Claim` type is the services internal model)

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/models/Claim.scala [hunk 5]
```diff
@@ -32,13 +31,12 @@ case class CreateClaimRequest(
...
-object CreateClaimRequest {
+object Claim {
 
-  implicit val format: OFormat[CreateClaimRequest] = Json.format[CreateClaimRequest]
+  implicit val format: OFormat[Claim] = Json.format[Claim]
 
-  def apply(id: String, userAnswers: UserAnswers): CreateClaimRequest =
-    new CreateClaimRequest(
-      userId = id,
+  def apply(id: String, userAnswers: UserAnswers): Claim =
```
Comment (crosslandwa) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/models/Claim.scala:38]: OBSERVATION: I think `id` is unused now (its not used in the creation of the `EISCreateCaseRequest`)

**Review by crosslandwa — DISMISSED** [review]

**Review by crosslandwa — APPROVED** [review]

**Pull request merged**